Allow update offsets based on timestep for extdata2g

darianboggs opened the pull request:
## Types of change(s)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)
- [ ] Refactor (no functional changes, no api changes)

## Checklist
- [ ] Tested this change with a run of GEOSgcm
- [x] Ran the Unit Tests (`make tests`)

## Description
Currently, the YAML configuration file for ExtData2 allows an offset for updates using an ISO8601 time duration string, such at `T09:00:00`. A user requested the capability to specify the offset as (+/-) timestep. This PR implements that capability. If map value for the `update_offset` is `HEARTBEAT`, the offset is set to an `ESMF_TimeInterval` equal to 1 timestep. If the map value is `-HEARTBEAT`, the offset is set to an `ESMF_TimeInterval` equal to -1 timestep. This may be extended to allow offsets of &#177;`n timestep`, where is a natural number.

## Related Issue
resolve #3064

bena-nasa commented on gridcomps/ExtData2G/ExtDataUpdatePointer.F90:
```diff
@@ -71,22 +88,72 @@ subroutine create_from_parameters(this,update_time,update_freq,update_offset,tim
+
+   end subroutine split_on
+      
+   function to_upper(s) result(u)
```
ESMF has a built in function to upper case strings so you can just use that.

darianboggs replied:
Thanks. Done.

darianboggs commented:
@bena-nasa and @tclune I added two tests to check if the result from using `HEARTBEAT` (string) matches the result from using ISO8601 Duration (string). They match for positive and negative `HEARTBEAT`.

Also, I added the capacity to include a `+` in front of the `update_offset` string. With the capability to use `HEARTBEAT`, I forsee a user using `+HEARTBEAT`, and I'd rather fix it now.
